src/amber/cli/templates/app/README.md.ecr

```diff
@@ -1,52 +1,84 @@
 # <%= @name %>
 
-Your description here
+A new project using [Amber Framework](https://amberframework.org/)
 
 ## Installation
 
+Install back-end and front-end dependencies.
+  
+```
+shards install
+npm install
+```
+
 Configure the `config/environments/development.yml` and set the `database_url` with your credentials to your database.
 
 Then:
 
-```shellsession
-shards update
+```
 amber db create migrate
 ```
 
 ## Usage
 
-To run the demo:
+### Development
+
+To build crystal files:
+
+```
+amber watch
+```
+
+To build assets:
 
-```shellsession
-$ shards build src/<%= @name %>.cr
-./<%= @name %>
 ```
+npm run watch
+```
+
+### Production
+
+To build a production release:
+
+```
+AMBER_ENV=production
+shards build --production --release <%= @name %>
```

Comment:
Maybe we can use production automatically if `--release` flags is used? WDYT?

```
# config/application.cr
{% if flag :release %}
settings.env = "production"
{% end %}
```

Reply:
I'd rather not, that defeats the purpose of the environment variable.

Reply:
Sorry, what does `--release` do in `shards build` command? I haven't seen it as a valid flag:

```sh
❯ shards --version                                                                                                                                                                     
Shards 0.7.1 (2017-10-12)
❯ shards build -h                                                                                                                                                                        
shards [options] <command>

Commands:
    build [targets] [options]
    check
    init
    install
    list
    prune
    update

Options:
    --no-color
    --version
    --production
    -v, --verbose
    -q, --quiet
    -h, --help

```

Reply:
> I'd rather not, that defeats the purpose of the environment variable.

@robacarp Yeah I know, that's Why I asked,

Sometimes I just think compiling amber projects for production is so repetitive

```
# Enable production env to read production.enc
AMBER_ENV=production

 # --production does not install DevDependencies
shards insttall --production

# --release enable crystal build optimizations for production
shards build --release <%= @name %>
```

Maybe we should add something like `amber build -e production`

We already have `amber run -e production` but it runs the binary by default, so I think this is more useful during development.

Of course I can do something like `AMBER_ENV=production shards build myproject` at works (unoptimized). I just thought maybe we can take advantage from `shards --production` and `crystal --release` flags.

> Builds an executable with --release flag?

@veelenga Yeah, `--release` builds executable with release flag, seems to be not documented yet

Reply:
Hm, I believe `shards build --release` does check dependencies for you, no need to do `shards install` beforehand.

I'd like to avoid making the way amber uses the underlying crystal commands and libraries too opaque, because it handicaps people when they need to wander outside the bounds of an ordinary Amber application. Perhaps a command like `amber build -e production` could be more transparent by echoing out what it's doing as it happens?

Reply:
> Hm, I believe shards build --release does check dependencies for you, no need to do shards install beforehand.

Yeah, you can use `shards build --production --release`

> Perhaps a command like amber build -e production could be more transparent by echoing out what it's doing as it happens?

Yeah I like that 👍

Reply:
> ...echoing out what it's doing as it happens?

We should do the same with `amber run` and `amber watch` using some kind of flag like `--verbose`

Reply:
Well, seems that `amber build` is useless, we already have `shards build`, currently I'm thinking on remove `amber run` too